### PR TITLE
Everfull Mug

### PR DESCRIPTION
**Pull request by SpringSkipper**
## What Does This PR Do
Adds a new, fairly simple healing item for wizards: the Everfull Mug!
With a new sprite designed by yours truly, the Everfull mug is available for the cost of 1 SP. Its primary practical use is that it can fill itself with omnizine at-will, with no cooldown. The other options aren't especially useful, and are there for whimsy and/or soul to give it that 'wizard' feel.

The full list of options is...
Omnizine
Ale
Wine
Holy Water
Welding Fuel
Vomit

I'm open to changing these or potentially adding more options based on feedback. I'm aiming for all of these other than omnizine to be fairly simplistic and weird, without necessarily being powerful.

## Why It's Good For The Game
The wizard feels lacking in a simple, straightforward, long-term healing option. They are forced to go to medbay at roundstart or pick one of the small number of gimmicky healing options. This would let wizards pay a smaller amount of SP to not be completely up shit creek if they feel like creeping around in maintenance for a while rather than revealing themself right out of the gate by poofing into scichem or medical storage.

_**But wizards shouldn't have unlimited healing!**_
Utility Holoparasite, Cursed Heart, Healing Apprentice, and Lich+Milk already exist, and this has clear tradeoffs when compared with those. Omnizine is a potent healing chemical, but it has the drawbacks of overdose and addiction. Further this does nothing to fix broken bones, internal bleeds, or other similar injuries.

_**It should cost 2 SP!**_
Comparing this to 2 SP items like wand belt, soul shards, Mjollnir, battle armor, necrostone, or scrying orb makes it clear to me that 2 SP would be horrifically overcosted without substantial additional utility, which I think would defeat the purpose of this being a fairly unassuming but neat trinket like its D&D equivalent. It feels like a much better fit alongside 1 SP items like meathook, cursed katana, warp cubes, or cursed heart.

## Testing
Spawned in, bought the mug through the spellbook, tried refilling it multiple times while doing all the drink-glass-related things with it I could think of, drank as much vomit as I desired.

## Changelog
:cl:
add: Everfull Mug, a new healing item for the wizard.
/:cl:

### CODE REVIEW

**Comment by Coolrune206** [comment]
Maybe add Mugwort to it? They get their little mug for free, but this costs spell-points and Mugwort is _very_ Wizard-y.

File: code/game/gamemodes/wizard/artefact.dm [hunk 1]
```diff
@@ -819,3 +819,38 @@ GLOBAL_LIST_EMPTY(multiverse)
...
+									"Wine" = "wine",
+									"Holy Water" = "holywater",
+									"Welder Fuel" = "fuel",
+									"Vomit" = "vomit",)
+	var/static/list/options_to_descriptions = list("Omnizine" = "a strange pink-white liquid",
+												"Ale" = "foamy amber ale",
+												"Wine" = "deep red wine",
+												"Holy Water" = "sparkling clear water",
+												"Welder Fuel" = "a dark, pungent, oily substance",
+												"Vomit" = "warm chunky vomit",)
```
Comment (Henri215) [inline code/game/gamemodes/wizard/artefact.dm:846]:
```suggestion
												"Vomit" = "warm chunky vomit")
```
End of lists dont need that

**Comment by SpringSkipper** [comment]
> Maybe add Mugwort to it? They get their little mug for free, but this costs spell-points and Mugwort is _very_ Wizard-y.
I wrestled with that idea. Omnizine heals faster but is overall weaker due to the addiction and overdose limit, as well as not purging chemicals. I'm not against replacing the omnizine with mugwort if people think that's reasonable, or just having both.

File: code/game/gamemodes/wizard/artefact.dm (hunk 1 above)
Comment (SpringSkipper) [reply code/game/gamemodes/wizard/artefact.dm:846]: I copy/pasted the format from how spellblade does it, and I was afraid to change that bit : P
I'll remove that from the lists then.

**Comment by Sayrbee** [comment]
if the mug can magically refill with multiple different chems then it should be more of an alchemy jug.

**Review by GDNgit — CHANGES_REQUESTED** [review]
Sanity checks

**Review by GDNgit — CHANGES_REQUESTED** [review]
why is git such an weird piece of shit to use sometimes ugh gimme a sec
And it reviewed it again?

**Comment by SpringSkipper** [comment]
> if the mug can magically refill with multiple different chems then it should be more of an alchemy jug.

It's based on/named after a combination of this item from D&D: https://dnd.arkalseif.info/items/magic-item-compendium--73/everfull-mug--396/index.html
... and this item from a D&D webcomic: https://www.goblinscomic.com/comic/04242006

But you make a fair point. An alchemy jug with a more useful variety of chemicals could be a potential step up if the balance folks feel this could stand to be more powerful.

**Comment by Gatchapod** [comment]
I was generally against the idea on Discord. After some further consideration, I think the item would be good, as long as it does not provide infinite mugwort.
**However**, the price. Yes, I am going to argue that.
Apprentice/necromantic stone and a staff of healing cost 3 SP. Infinite.
Charge and belt of wands cost 3 SP. Finite, but also providing offense and utility.
Bind soul costs 2 SP. Semi-infinite.
Support guardian costs 2 SP. Infinite.
Voice of god costs 2 SP. Technically infinite, but who uses it for healing?
Cursed heart costs 1 SP. Infinite, but also very risky.

The only 1 SP infinite healing option can screw you up badly if you misplay. However, making it 2 SP would likely make it the worst healing option bar VoG. Now, to compensate for that, we could add pump up to the list of available chemicals, which would help wizard shrug off stamina damage. I think this would be a very fair deal for 2 SP, then.
Or just go fuck it and give wizards meth ¯\_(ツ)_/¯

**Comment by SpringSkipper** [comment]
> I was generally against the idea on Discord. After some further consideration, I think the item would be good, as long as it does not provide infinite mugwort. **However**, the price. Yes, I am going to argue that. Apprentice/necromantic stone and a staff of healing cost 3 SP. Infinite. Charge and belt of wands cost 3 SP. Finite, but also providing offense and utility. Bind soul costs 2 SP. Semi-infinite. Support guardian costs 2 SP. Infinite. Voice of god costs 2 SP. Technically infinite, but who uses it for healing? Cursed heart costs 1 SP. Infinite, but also very risky.
> 
> The only 1 SP infinite healing option can screw you up badly if you misplay. However, making it 2 SP would likely make it the worst healing option bar VoG. Now, to compensate for that, we could add pump up to the list of available chemicals, which would help wizard shrug off stamina damage. I think this would be a very fair deal for 2 SP, then. Or just go fuck it and give wizards meth ¯\_(ツ)_/¯

I'm going into a lot of detail here; partly for any other balance folks who are interested in giving it a good lookover.

I don't count wand belt since, as you point out, it isn't infinite and has so much other utility. Apples to oranges.
Voice of God can't affect yourself; It can only heal others.

I'll cover the other options, but first I'm going to remind you that omnizine heals 2 brute/burn and 1 tox/oxyloss per life cycle (i.e. every 2 seconds).

**-Healing Staff-**
Healing apprentice comes with their own staff, so 2 SP. Maybe another 1 SP for recall object or warp cubes to ensure access, but those come with their own benefits.
Necrostone + Healing staff is 3 SP, but also includes the ability to make your own hijack-level minions with hands and the ability to superheal with milk. Don't underestimate that.

The obvious reason to go for the healing staff is that it is the most powerful. The staff fixes everything. Everything.
**Bonebreaks**, **IBs**, radiation-induced mutations, diseases, brain damage, **shock/heart failure**, **embeds**, missing organs, **memechems**, appendicitis, legcuffs, being hungry... I went ahead and bolded all the extremely relevant and potentially round-ending ones that come up in more wizard rounds than not. The apprentice even comes with the added bonus that the round doesn't end if you die but they survive, meaning they can fix you dying as well if they get a chance. Yes, there is the added logistics of needing to keep the other person around in order to use the staff on you, but that also comes with the benefit of a second pair of hands and the healing is instant and absolute.

**-Cursed Heart-**
Cursed Heart, you either know how to use it or you don't. If you know how to use it then, barring falling asleep to N2O or memechems, you're generally pretty safe from dying to it. The risk can accurately be described as one of experience; very comparable to accidentally ODing on omnizine. The greater cost is the amount of attention the heart requires; it is annoying to maintain, and is especially constraining when it comes to talking, getting involved in melee combat, using more nuanced or complex spells, or interacting with complex systems (say, hacking the AI or ordering on a supply console).

The issue I have with the constraints of heart is that it does not actually much limit how _powerful_ your build can be, since a lot of the best options (forcewall, jaunt, rod form, soulstone belt, lightning bolt, bluespace spellblade, einath...) require minimal input and comparatively little attention. The way it skews you into picking an extremely simple build is a large part of the reason for me making this PR in the first place. It is a tradeoff, but perhaps not the specific tradeoff that you think it is.

As for actual healing, the heart heals 25 brute, burn, and oxyloss every 6 seconds. This is 8.33 per 2 seconds, which is four times more brute/burn and eight times more oxyloss than omnizine. Importantly, it lets you spacewalk unassisted while omnizine isn't quite able to keep up with the burn damage caused by the cold of space.

Side note: cursed heart has protections against lag and disconnects. It shouldn't kill you if you lag or your client crashes.

**-Utility Holoparasite-**
This is probably the most direct point of comparison if you were to up the cost to 2 SP. The utility holoparasite can heal 5 Brute, Burn, Tox, and Oxy every 2 seconds (they can click to heal you every 0.8 seconds, however the heal has a hidden cooldown of 2 seconds where it will **say** they heal you but you don't actually heal at all, with no indication that you didn't heal. Spaghetti code indeed). Lets say for argument's sake that they just spam-click you, and you end up only healing every 2.5 seconds. That's still twice as fast as omnizine. This also comes with the benefit of a second body that takes less damage than you (unless you have armor. Also worth noting that they always transfer damage as brute damage, which means that if some lasers hit you and some lasers hit them then they get double value by healing your brute and burn damage at the same time.) They can also bodyblock, drag you around when you're stunned, and attack for a serviceable 15 damage not to mention the ability to teleport people which is a decent transportation option, or a hilarious way to kill people if you're creative. This is almost the definition of twice as useful as the everfull mug.

Yes, your holoparasite can be new. But they shouldn't be disobeying any direct orders you give them. They know the holopara type before they join in, so you shouldn't get anyone who loathes the idea of playing your healer. Even the baldest of baldies understands the idea of spam-clicking you on heal mode, and you have time in the wizard den to explain how the warp ability works. Finally, you can re-roll your ghost if they're just unworkably bald, constantly AFK, or completely unwilling to listen. Compared to the risks of ODing on omnizine, this is a pretty mild concern.

**-Lich+Milk-**
This is a tricky comparison; milk is readily available, but if you consider that then you also need to take into consideration the whole thing of just poofing into med storage at roundstart. I'm going to go on record as saying I think cursed heart and utility guardian are better comparison points.

That said, Lich makes you a skeleton. This means immunity to chems, pressure, temperature, atmospherics, heart failure, and bleeding. For that, you pay the cost of oldcrit which substantially reduces your effective maximum health compared to the other healing options.

Milk is easily available, and each tick heals you 4 brute and burn damage; twice as fast as omnizine. Further, it has a chance every single tick to regenerate missing limbs and fix broken bones; two of the few lingering injuries that skeletons aren't immune to. A few bottles, beakers, or cartons full of milk should mean that you're set for the full round.

Obviously, Lich also lets you revive when you die. That comes with its own set of caveats though, and you're likely to have more trouble finding milk the second time around once the crew is wise to you than the first. For sake of comparison I'm not going to count it, and just assume that your phylactery is in your backpack or pocket.

**Comment by jughu** [comment]
eww wizard balance wizards can be instantly killed

**Review by Sirryan2002 — APPROVED** [review]

**Comment by Peakwastaken** [comment]
This is a very interesting idea !

I personally think it'd be fine to also have mugwort in it, more on brand and it's much weaker than omnizine with the only upside of being able to purge chemicals.

Can the mug's content be changed after picking it the first time ? If yes is there a cooldown to that ? Does it refill instantly so you can drink at infinitum, or is it a slow refill ? Can you splash it ? 

I'd even venture to say you could add a bigger list of reagent, maybe from the synthesizable list - depending on how the mug functions I think it could be fair as omni is already one of the strongest chems you can find out there. (Obviously if there's no refill timer and you can spam splash synthflesh on yourself, or keep switching between meth/mannitol, for example, it'd be a problem)

**Comment by SpringSkipper** [comment]
> I personally think it'd be fine to also have mugwort in it, more on brand and it's much weaker than omnizine with the only upside of being able to purge chemicals.
> **(...)**
> I'd even venture to say you could add a bigger list of reagent, maybe from the synthesizable list - depending on how the mug functions I think it could be fair as omni is already one of the strongest chems you can find out there. (Obviously if there's no refill timer and you can spam splash synthflesh on yourself, or keep switching between meth/mannitol, for example, it'd be a problem, or more suited for 2 sp)

As it's finally marked 'awaiting merge' I'd rather let it get merged in as-is. However if I notice that it seems too weak, I won't be shy about improving it. The main value of the item should be unlimited slow-acting healing chemicals, as a more straightforward alternative to cursed heart or the healing holoparasite, and I'm hesitant to branch it out into more powerful pyrotechnics, toxins, antistun drugs, or more complex healing medicines unless it proves too ineffective without them.

I also don't want it to make elaborate meta chemical cocktails - particularly highly-synthetic ones - since it goes against the concept that this is an old timey magical mug that makes things wizards are familiar with. Omnizine in its flavor descriptions is usually described as quasi-magical or being 'life itself'. Welding fuel is something that is commonly available and I think fits the bill for something a wizard might come up with when they think "I want something flammable!", though it could maybe be changed to phlogiston since that itself is a direct D&D reference to a magical substance. But things like a saline/epi/salbutamol anticrit cocktail or a meth/mannitol/sugar mixture feels too gamey, arbitrary, and complex to fit the flavor. I could see something like "herbal poultice" being a mix of morphine, bicardine, and kelotane maybe. Botany drugs would be a much better fit than chem dispenser drugs.

> Can the mug's content be changed after picking it the first time ? If yes is there a cooldown to that ? Does it refill instantly so you can drink at infinitum, or is it a slow refill ? Can you splash it ?

It can be changed any number of times, instantly replacing all contents and refilling to full with no cooldown. It is a drink container so you can't splash it, but you can pour it into a separate beaker and then splash that.

**Comment by Peakwastaken** [comment]
Oh I totally agree, if it can refill instantly it's better suited the way it is and still works great !

**Review by Sirryan2002 — APPROVED** [review]

**Review by farie82 — APPROVED** [review]
Alchemy jug time

**Pull request merged**